technologies/job/aws-sagemaker/clone-trainingjob/instanceActions.js

```diff
@@ -0,0 +1,182 @@
+  try {
+    console.log('START INSTANCE:', instance);
+
+    if ( job.featuresValues.trainingjobname == null || job.featuresValues.trainingjobname.trim() === '' ) {
+          return Response.error('Training Job name is required.', new Error("Validation Error"));
+    }
+
+    if ( job.featuresValues.trainingjobinput == null || job.featuresValues.trainingjobinput.trim() === '' ) {
+          return Response.error('Training Job Input S3 location is required.', new Error("Validation Error"));
+    }
+
+    if ( job.featuresValues.trainingjoboutput == null || job.featuresValues.trainingjoboutput.trim() === '' ) {
+          return Response.error('Training Job Output S3 location is required.', new Error("Validation Error"));
+    }
+
+    AWS.config.update({credentials: { accessKeyId : job.featuresValues.endpoint.aws_access_key_id, secretAccessKey:  job.featuresValues.endpoint.aws_secret_access_key}});
+    AWS.config.update({region: job.featuresValues.endpoint.region});
+    
+    const sagemaker = new AWS.SageMaker({apiVersion: '2017-07-24'});
+    console.log('Get Job Details',  job.featuresValues.trainingjobs.label)
+    const data = await sagemaker.describeTrainingJob( params = {
+      TrainingJobName: job.featuresValues.trainingjobs.label
+    }).promise();
+
+    console.log("Retrieved Template Job Details");
+    console.log(data);
+
+    console.log("Create New Job based on template parameters");
+    var jparams = {
+    AlgorithmSpecification:{
+     TrainingInputMode: data.AlgorithmSpecification.TrainingInputMode,
+     AlgorithmName: data.AlgorithmSpecification.AlgorithmName,
+     EnableSageMakerMetricsTimeSeries: data.AlgorithmSpecification.EnableSageMakerMetricsTimeSeries,
+     TrainingImage: data.AlgorithmSpecification.TrainingImage
+    },
+    OutputDataConfig: {
+        S3OutputPath: job.featuresValues.trainingjoboutput
+      },
+    ResourceConfig: data.ResourceConfig,
+    RoleArn: data.RoleArn,
+    StoppingCondition: data.StoppingCondition,
+    TrainingJobName: job.featuresValues.trainingjobname,
+    DebugHookConfig: data.DebugHookConfig,
+    DebugRuleConfigurations: data.DebugRuleConfigurations,
+    EnableInterContainerTrafficEncryption: data.EnableInterContainerTrafficEncryption,
+    EnableManagedSpotTraining: data.EnableManagedSpotTraining,
+    EnableNetworkIsolation: data.EnableNetworkIsolation,
+    ExperimentConfig: data.ExperimentConfig,
+    HyperParameters: data.HyperParameters,
+    InputDataConfig: [
+        {
+          ChannelName: 'train', /* required */
+          DataSource: { /* required */
+            S3DataSource: {
+              S3DataType: 'S3Prefix',
+              S3Uri:  job.featuresValues.trainingjobinput,
+              S3DataDistributionType: 'FullyReplicated',
+            }
+          },
+          CompressionType: 'None',
+          InputMode: 'File'
+        }
+    ],
+    Tags: data.Tags
+    }
+
+    const clone=await sagemaker.createTrainingJob( params = jparams ).promise();
```

Comment:
```suggestion
    const clone=await sagemaker.createTrainingJob(jparams).promise();
```

No need to reassign to another variable.

Reply:
Done